### PR TITLE
[#154335393] GetService in public API

### PR DESCRIPTION
**Pull request by gunzip**

### CODE REVIEW

File: api/definitions.yaml [hunk 1]
```diff
@@ -241,6 +241,28 @@ Service:
...
+    service_id:
+      $ref: "#/ServiceId"
+    service_name:
+      $ref: "#/ServiceName"
+    organization_name:
+      $ref: "#/OrganizationName"
+    department_name:
+      $ref: "#/DepartmentName"
+    version:
+      type: integer
```
Comment (cloudify) [inline api/definitions.yaml]: is it possible to define a non-negative integer with swagger constrainta?

File: api/definitions.yaml [hunk 2]
```diff
@@ -241,6 +241,28 @@ Service:
...
+    id:
```
Comment (cloudify) [inline api/definitions.yaml]: the actual document ID is redundant as it is composed of `service_id` and `version` and should be used internally only

File: api/public_api_v1.yaml [hunk 3]
```diff
@@ -9,6 +9,33 @@ schemes:
...
+      summary: Get Service
+      description: A previously created service with the provided service ID is returned.
+      responses:
+        '200':
+          description: Service found.
+          schema:
+            "$ref": "#/definitions/ServicePublic"
+          examples:
+            application/json:
+              id: 2b3e728c1a5d1efa035c-0000000000000001
```
Comment (cloudify) [inline api/public_api_v1.yaml]: see my comment above

File: lib/api/admin_api.ts [hunk 4]
```diff
@@ -497,6 +497,25 @@ export const specs = {
         "authorized_recipients"
       ]
     },
+    ServicePublic: {
+      title: "Service (public)",
+      description: "A Service tied to an user's subscription.",
```
Comment (cloudify) [inline lib/api/admin_api.ts]: `tied` has a literal meaning (having one's hands tied), perhaps `associated` would be better in this context

File: lib/api/admin_api.ts [hunk 5]
```diff
@@ -497,6 +497,25 @@ export const specs = {
...
+      type: "object",
+      properties: {
+        service_id: { $ref: "#/definitions/ServiceId" },
+        service_name: { $ref: "#/definitions/ServiceName" },
+        organization_name: { $ref: "#/definitions/OrganizationName" },
+        department_name: { $ref: "#/definitions/DepartmentName" },
+        version: { type: "integer" },
+        id: { type: "string" }
```
Comment (cloudify) [inline lib/api/admin_api.ts]: see my comment above

File: lib/controllers/services.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+/**
+ * Converts a retrieved service to a service that can be shared via API
+ */
+function retrievedServiceToPublic(
+  retrievedService: RetrievedService
+): ApiService {
+  return {
+    department_name: retrievedService.departmentName,
+    id: retrievedService.id,
```
Comment (cloudify) [inline lib/controllers/services.ts]: see my comment above

File: lib/controllers/services.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+/**
+ * Wraps a GetService handler inside an Express request handler.
+ */
+export function GetService(serviceModel: ServiceModel): express.RequestHandler {
+  const handler = GetServiceHandler(serviceModel);
+  const azureUserAttributesMiddleware = AzureUserAttributesMiddleware(
+    serviceModel
+  );
+  const middlewaresWrap = withRequestMiddlewares(
+    AzureApiAuthMiddleware(new Set([UserGroup.ApiServiceRead])),
```
Comment (cloudify) [inline lib/controllers/services.ts]: I think we should differentiate the permissions for the admin endpoint and the public endpoint, we should have an `ApiPublicServiceRead` permission

File: api/definitions.yaml (hunk 1 above)
Comment (gunzip) [reply api/definitions.yaml]: yes but then we have to put this constraint on all the non negative integer fields which is probably beyond this PR scope

Comment (cloudify) [reply api/definitions.yaml]: agreed

**Pull request merged**